pgcli/main.py

```diff
@@ -653,6 +664,19 @@ def obfuscate_process_password():
 
     setproctitle.setproctitle(process_title)
 
+def output_fits_screen(output, screen_size):
+    for entry in output:
+        new_line_index = entry.find("\n")
+        if new_line_index == -1:
+            if len(entry) > screen_size.columns:
+                return False
+        else:
+            if new_line_index > screen_size.columns:
+                return False
+            if entry.count("\n") >= screen_size.rows:
```

Comment:
Usually an output looks like this after it has been formatted: 
` ['+------+-------+\n|   id | key   |\n|------+-------|\n|    1 | foo   |\n+------+-------+', 'SELECT 1']`

 (I check after it has been tabulated because tabulation may alter it based on padding or `\x auto`)

Now it occurs to me that this check will not work if output has more than the standard two entries though.. In that case `count("\n")` per entry should be added to a total and checked at the end of the for loop to see whether it exceeds `screen_size.rows`.